src/Dns/Dns.Test/ScenarioTests/ZoneTests.ps1

```diff
@@ -80,6 +80,59 @@ function Test-ZoneCrud
 	Remove-AzResourceGroup -Name $resourceGroup.ResourceGroupName -Force
 }
 
+<#
+.SYNOPSIS
+Zone creation with delegation
+#>
+function Test-ZoneWithDelegation
+{
+	#Parent zone creation
+	$parentZoneName = Get-RandomZoneName
+    $resourceGroup = TestSetup-CreateResourceGroup
+	$createdParentZone = New-AzDnsZone -Name $parentZoneName -ResourceGroupName $resourceGroup.ResourceGroupName -Tags @{tag1="value1"}
```

Comment:
@ShwethaRaghupathi I would that this test is restructured in the following way:

```powershell
function Test-ZoneWithDelegation
{
    # Parent zone creation
    $parentZoneName = Get-RandomZoneName
    $resourceGroup = TestSetup-CreateResourceGroup
    try
    {
        # (Almost) all of the remaining test code here
    }
    finally
    {
        Remove-AzResourceGroup -Name $resourceGroup.ResourceGroupName -Force
    }
}
```

This structure allows you to do the necessary setup, run the cmdlets to be tested with the assertions in the `try` block, and at the end in the `finally` block, you delete the resource group containing all of the resources created during the test. This ensures that the resource group is _always_ deleted when the test is recorded and doesn't leave any additional resources hanging around.